src/components/Book.js

```diff
@@ -1,69 +1,74 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { useDispatch } from 'react-redux';
-import { removeBook } from '../redux/books/booksSlice';
+import { deleteBook, fetchBooks } from '../redux/books/booksSlice';
 import '../App.css';
+import styles from '../styles/Book.module.css';
 
-const Book = ({ book }) => {
+const Book = ({ booksListObj }) => {
   const dispatch = useDispatch();
-  const handleRemove = (id) => {
-    dispatch(removeBook(id));
+
+  const handleDelete = async (id) => {
+    await dispatch(deleteBook(id));
+    dispatch(fetchBooks());
   };
+
   return (
-    <div className="bookCard">
-      <div className="bookInfo">
-        <div className="mainInfo">
-          <p className="category">{book.category}</p>
-          <h1 className="title">{book.title}</h1>
-          <span className="author">{book.author}</span>
+    <div className={styles.bookCard}>
+      <div className={styles.bookInfo}>
+        <div className={styles.mainInfo}>
+          <p className={styles.category}>{booksListObj.category}</p>
+          <h1 className={styles.title}>{booksListObj.title}</h1>
+          <span className={styles.author}>{booksListObj.author}</span>
         </div>
-        <ul className="buttons">
+        <ul className={styles.buttons}>
           <li className="buttonItem">
-            <button type="button" className="button1">
+            <button type="button" className={styles.button1}>
               Comment
             </button>
           </li>
           <li className="buttonItem">
             <button
               type="button"
-              className="button"
-              onClick={() => handleRemove(book.id)}
+              className={styles.button}
+              onClick={() => {
+                handleDelete(booksListObj.item_id);
+              }}
             >
               Remove
             </button>
           </li>
           <li className="buttonItem">
-            <button type="button" className="button">
+            <button type="button" className={styles.button}>
               Edit
             </button>
           </li>
         </ul>
       </div>
 
-      <div className="completed">
-        <div className="progressCercle">
-          <p className="progressCompleted">
-            <span className="percentage">
-              {book.completed}
-              %
-            </span>
-            <span className="completedWord">Completed</span>
-          </p>
-        </div>
+      <div className={styles.completed}>
+        <div className={styles.progressCircle} />
```

Comment:
[REQUIRED] Kindly ensure that the size of your progress circle is similar to the template.  and it has a greyish colour for the section that does not have a blue background. This is a requirement for this milestone.

![image](https://github.com/AbdelazizSoliman/bookstore/assets/62938735/91804b72-7c5f-4b9d-8b45-00d9351d669b)